core-schema/src/main/resources/sql-schema/V1__Create_initial_schema.sql

```diff
@@ -47,10 +47,20 @@ CREATE TABLE prefs_flags (
   FOREIGN KEY (user_id) REFERENCES user_ids (id) ON DELETE CASCADE
 );
 
-CREATE TABLE prefs_blockedusers (
+CREATE TABLE friends_blocked_users (
   user_id INT NOT NULL,
   blocked_user_id INT NOT NULL,
   FOREIGN KEY (user_id) REFERENCES user_ids (id) ON DELETE CASCADE,
   FOREIGN KEY (blocked_user_id) REFERENCES user_ids (id) ON DELETE CASCADE,
-  INDEX (user_id)
+  INDEX (user_id),
```

Comment:
this index is redundant because user_id is the first column in the other index.
`UNIQUE (user_id, blocked_user_id)` - this allows fast access when you select according to user_id and blocked_user_id. It also allows fast access when you select against only user_id individually. If you needed to select against blocked_user_id individually, *then* you would want a separate `INDEX (blocked_user_id)`.

* A /listtheusersihaveblocked command - implementing this requires us to select against user_id. Therefore, `UNIQUE (user_id, blocked_user_id)` covers this use case.
* A /haveiblocked <otheruser> command - implementing this requires us to select against both user_id and blocked_user_id. Therefore, `UNIQUE (user_id, blocked_user_id)` covers this use case.
* A /listwhoisblockingme command - implementing this requires us to select against blocked_user_id. Therefore, `UNIQUE (user_id, blocked_user_id)` DOES NOT cover this use case, and we would want `INDEX (blocked_user_id)` to do that.

Since I do not think we will be implementing the last command, `INDEX (user_id)` is a waste. If we ever want a `/listwhoisblockingme`, we can add this index using another flyway migration.

Reply:
Your point is correct: Users shouldn't be able to see who is blocking them either, so that would be a waste.